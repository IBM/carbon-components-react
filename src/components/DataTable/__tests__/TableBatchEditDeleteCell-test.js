import React from 'react';
import { shallow } from 'enzyme';
import { TableBatchEditDeleteCell } from '../';

describe('DataTable.TableBatchEditDeleteCell', () => {
  it('should render', () => {
    const wrapper = shallow(<TableBatchEditDeleteCell />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom attributes on the top-level node', () => {
    const wrapper = shallow(<TableBatchEditDeleteCell data-foo="Foo" />);
    expect(wrapper.prop('data-foo')).toBe('Foo');
  });

  it('should render custom CSS class on the top-level node', () => {
    const wrapper = shallow(<TableBatchEditDeleteCell className="class-foo" />);
    expect(wrapper.hasClass('class-foo')).toBe(true);
  });

  it('should render custom child nodes', () => {
    const wrapper = shallow(
      <TableBatchEditDeleteCell>
        <div />
        <span />
      </TableBatchEditDeleteCell>
    );
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('span').length).toBe(1);
  });
});
