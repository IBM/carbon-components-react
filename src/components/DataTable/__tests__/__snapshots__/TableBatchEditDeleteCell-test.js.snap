// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableBatchEditDeleteCell should render 1`] = `
<TableCell
  className="bx--data-table-v2--batch-edit-delete-cell"
/>
`;
